src/test/java/com/google/graphgeckos/dashboard/storage/DatastoreRepositoryTests.java

```diff
@@ -209,4 +218,33 @@ public void testGetLastRevisionsQueryAfterDeletion() throws IOException, Interru
     Assert.assertEquals(results.get(0), dummy2);
     Assert.assertEquals(results.get(1), dummy1);
   }
+
+  @Test
+  public void testValidGetAndSetIndexRequests() throws IOException, InterruptedException {
```

Comment:
This tests uses setBuildbotIndex and getBuildbotIndex at once. It would be better to test them separately and inspect / prepare the emulator instead of using both methods. This allows for isolated testing. Please consider this in case the emulator allows for this.

Testing at least with 2 different value pairs shows that you can use different values / store multiple things. You could for example call the the setter and getter multiple times in the same test.

Reply:
The emulator allows only for blank states at the beginning of each test. That being said, I could directly use the instance of Datastore that I provide to the DatastoreRepository to create the entities I need. Should I apply this to all other tests, since many of them depend on multiple methods at the same time?

Reply:
Yes, I think that would be good! The more we can isolate the newly tested functionality, the better.